### PR TITLE
Make setTracks work via the API

### PR DESCRIPTION
**Pull request by markspolakovs**
We can't pass in MyRadio_Track objects via the API, so let it accept IDs.

### CODE REVIEW

File: src/Classes/iTones/iTones_Playlist.php [hunk 1]
```diff
@@ -366,9 +366,10 @@ public function validateLock($lockstr)
      *
      * Once that's done, go over every Track still in the temporary list and remove them from the Playlist
      *
-     * @param MyRadio_Track[] $tracks  Tracks to put in the playlist.
-     * @param string          $lockstr String that provides Write access to this Playlist. Acquired from acquireLock();
-     * @param string          $notes   Optional. A textual commit message about the change.
+     * @param MyRadio_Track[]|int[] $tracks  Tracks to put in the playlist.
+     * @param string                $lockstr String that provides Write access to this Playlist. Acquired from acquireLock();
+     * @param string|null           $notes   Optional. A textual commit message about the change.
+     * @param MyRadio_User|null     $user   User to attribute the update to
```
Comment (mstratford) [inline src/Classes/iTones/iTones_Playlist.php:372]: How is the MyRadio_User decided here for the API? We should change it to use whoever the logged in user is, instead.

Comment (mstratford) [reply src/Classes/iTones/iTones_Playlist.php:373]: Hum, I see it actually already does this, I presume the API won't let you feed in a user either, but why can you send a user in anyway?

Comment (markspolakovs) [reply src/Classes/iTones/iTones_Playlist.php:373]: Look a few lines down, it already does that :) Maybe we should just get rid of `$user` though, as it doesn't  appear to be used anywhere and could pose some issues were this exposed to the API. However, https://github.com/UniversityRadioYork/MyRadio/blob/master/src/Classes/Daemons/MyRadio_PlaylistsDaemon.php#L76-L80 uses it...

Comment (markspolakovs) [reply src/Classes/iTones/iTones_Playlist.php:373]: Fairly sure we can trivially fix this by changing this to use `getCurrentOrSystemUser()` and removing the `$user` param.

Comment (mstratford) [reply src/Classes/iTones/iTones_Playlist.php]: Yeah, that's what I was thinking 👍

**Review by mstratford — APPROVED** [review]
LGTM

**Pull request merged**